# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1466533425 14400
#      Tue Jun 21 14:23:45 2016 -0400
# Node ID 0de2c283962301af4aa0aa42ac075bc257c6087f
# Parent  307a2d7ff29b5f1e60643a7692837050ac2241ff
* defaults.cc (subst_octave_home): String length is size_t.

diff --git a/libinterp/corefcn/defaults.cc b/libinterp/corefcn/defaults.cc
--- a/libinterp/corefcn/defaults.cc
+++ b/libinterp/corefcn/defaults.cc
@@ -100,17 +100,17 @@ subst_octave_home (const std::string& s)
   std::string retval;
 
   std::string prefix = OCTAVE_PREFIX;
 
   retval = s;
 
   if (Voctave_home != prefix)
     {
-      octave_idx_type len = prefix.length ();
+      size_t len = prefix.length ();
 
       if (s.substr (0, len) == prefix)
         retval.replace (0, len, Voctave_home);
     }
 
   if (octave::sys::file_ops::dir_sep_char () != '/')
     std::replace (retval.begin (), retval.end (), '/',
                   octave::sys::file_ops::dir_sep_char ());
